platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py

```diff
@@ -46,17 +49,23 @@ class Eeprom(eeprom_tlvinfo.TlvInfoDecoder):
     EEPROM_DECODE_OFFSET = 0
     EEPROM_DECODE_CONTENT = 2
 
+    def _wait_for_eeprom_symlink(self):
+        if 'simx' not in self.onie_platform:
+            for attempt in range(self.RETRIES):
+                if not os.path.islink(EEPROM_SYMLINK):
+                    time.sleep(1)
+                else:
+                    break  
+        
+            if not (os.path.exists(EEPROM_SYMLINK) \
+                    or os.path.isfile(os.path.join(CACHE_ROOT, CACHE_FILE))):
+                log_error("Nowhere to read syseeprom from! No symlink or cache file found")
+                raise RuntimeError("No syseeprom symlink or cache file found")
+        
     def __init__(self):
-        for attempt in range(self.RETRIES):
-            if not os.path.islink(EEPROM_SYMLINK):
-                time.sleep(1)
-            else:
-                break  
-
-        if not (os.path.exists(EEPROM_SYMLINK) \
-                or os.path.isfile(os.path.join(CACHE_ROOT, CACHE_FILE))):
-            log_error("Nowhere to read syseeprom from! No symlink or cache file found")
-            raise RuntimeError("No syseeprom symlink or cache file found")
+        self.onie_platform = get_platform()
```

Comment:
@tomer-israel please make this variable private, e.g. `self._onie_platform = get_platform()`